scrapper/sources/greenhouse.py

```
@@ -0,0 +1,390 @@
+    MAX_CONCURRENT = 5
+
+    # Shared across instances to avoid repeated list fetches.
+    _job_list_cache: Dict[str, Tuple[datetime, List[dict]]] = {}
+    _cache_lock = asyncio.Lock()
+
+    @classmethod
+    def clear_cache(cls) -> None:
+        """Clear cached job lists (useful for tests and cache invalidation)."""
+        cls._job_list_cache.clear()
+    
+    def __init__(self):
```

Ruff (0.15.12)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)